fairseq/utils.py

```diff
@@ -346,7 +346,9 @@ def log_softmax(x, dim: int, onnx_trace: bool = False):
 def get_perplexity(loss, round=2, base=2):
     if loss is None:
         return 0.
-    return safe_round(base**loss, round)
+    if torch.is_tensor(loss):
+        loss = loss.cpu()
+    return safe_round(np.power(base, loss), round)
```

Comment:
We're trying to reduce the number of GPU-to-CPU transfers for performance reasons. Does something like this work:
```python
try:
    return safe_round(base**loss, round)
except OverflowError:
    return float('inf')
```